packages/svelte/src/internal/client/proxy.js

```diff
@@ -68,7 +84,7 @@ export function proxy(value, immutable = true) {
  * @returns {Record<string | symbol, any>}
  */
 function unwrap(value, already_unwrapped = new Map()) {
-	if (typeof value === 'object' && value != null && !is_frozen(value) && STATE_SYMBOL in value) {
+	if (typeof value === 'object' && value != null && STATE_SYMBOL in value) {
```

Comment:
Do we really even need a `STATE_SYMBOL` in the value? Then we can unwrap derived proxied objects too.